src/core/pointcloud/qgseptpointcloudindex.cpp

```diff
@@ -279,20 +280,37 @@ QgsPointCloudBlock *QgsEptPointCloudIndex::nodeData( const IndexedPointCloudNode
   if ( !found )
     return nullptr;
 
+  // create a copy of the expression to pass to the decoder
+  QgsPointCloudExpression filterExpression = mFilterExpression;
```

Comment:
We can't use the same expression object for every call to the decoder as it might be concurrently accessed for different nodes eg. from 2d and 3d view.
An alternative could be to pass the expression by value to the decoder, and let `prepare()` do the detach.